Now that we have cleaned the data by player names we have to clean it by year. We start by loading the datasets.

In [9]:
# We start by importing our data sets
import pandas as pd
import numpy as np

salaries = pd.read_csv("data/NBA_season1718_salary.csv")
playerData1 = pd.read_csv("data/Players_2.csv")
playerData2 = pd.read_csv("data/Seasons_Stats_2.csv")
playerData3 = pd.read_csv("data/player_data_2.csv")

salaries.head()
playerData1.head()
playerData2.head()
playerData3.head()

name  year_start  year_end position height  weight  \
0   Alex Abrines        2017      2018      G-F    6-6   190.0   
1     Quincy Acy        2013      2018        F    6-7   240.0   
2   Steven Adams        2014      2018        C    7-0   255.0   
3  Arron Afflalo        2008      2018        G    6-5   210.0   
4  Alexis Ajinca        2009      2017        C    7-2   248.0   

         birth_date                                college  
0    August 1, 1993                                    NaN  
1   October 6, 1990                      Baylor University  
2     July 20, 1993               University of Pittsburgh  
3  October 15, 1985  University of California, Los Angeles  
4       May 6, 1988                                    NaN

The playerData1 dataframe does not have a date column, so we will not clean it here. Seasons_Stats_2 has a column labeled "Year" so we will clean it and only leave the ones that have data from 2017 or after, as these are the years we have salaries data for. Seasons_Stats_2 has a start year and an end year column, so we will only filter for those rows that contain one of the two desired years in their scope.

In [10]:
finalPlay2 = playerData2[playerData2['Year'].isin([2017, 2018])]
print(finalPlay2.shape)
finalPlay2.to_csv("data\final_Season_Stats.csv")

(506, 53)


This means that there are 506 rows that have the desired players and years in the Seasons_Stats_2 dataset. We proceed by doing a similar thing for the player_data_2.

In [11]:
finalPlay3 = playerData3[(playerData3['year_start'] <= 2018) & (playerData3['year_end'] >= 2017)]
print(finalPlay2.shape)
finalPlay3.to_csv("data\final_player_data.csv")

(506, 53)


This means that both of the files have exactly 506 "desireable" rows. The next thing we want to do is combine all the data sets into one. Lets start by checking if the players in the two "final" data sets are the same amongst the two data sets and unique within them.